CalorieTracker/CalorieTracker/ViewControllers/IntakeTableViewController.swift

```diff
@@ -0,0 +1,186 @@
+        fetchRequest.sortDescriptors = [NSSortDescriptor(key: "timestamp", ascending: true)]
+        let moc = CoreDataStack.shared.mainContext
+        let frc = NSFetchedResultsController(fetchRequest: fetchRequest, managedObjectContext: moc, sectionNameKeyPath: nil, cacheName: nil)
+        frc.delegate = self
+        do {
+            try frc.performFetch()
+        } catch {
+            NSLog("Unable to fetch intakes from main context: \(error)")
+        }
+        return frc
+    }()
+    
+    // MARK: - View Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        formatter.dateStyle = .medium
+        formatter.timeStyle = .short
+        NotificationCenter.default.addObserver(self, selector: #selector(updateChart), name: .shouldUpdate, object: nil)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        tableView.reloadData()
+        updateChart()
+    }
+
+    // MARK: - Table view data source
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        fetchedResultsController.sections?[section].numberOfObjects ?? 0
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "intakeCell", for: indexPath)
+        let date = formatter.string(from: fetchedResultsController.object(at: indexPath).timestamp ?? Date())
+        cell.textLabel?.text = "Calories: \(fetchedResultsController.object(at: indexPath).calories)"
+        cell.detailTextLabel?.text = date
+        return cell
+    }
+
+    override func tableView(_ tableView: UITableView,
+                            commit editingStyle: UITableViewCell.EditingStyle,
+                            forRowAt indexPath: IndexPath) {
+        if editingStyle == .delete {
+            let intake = fetchedResultsController.object(at: indexPath)
+            intakeController.deleteIntakeFromServer(intake) { _ in
+                let moc = CoreDataStack.shared.mainContext
+                moc.delete(intake)
+                do {
+                    try moc.save()
+                } catch {
+                    moc.reset()
+                    NSLog("Error saving managed object context: \(error)")
+                }
+                DispatchQueue.main.async {
+                    self.updateChart()
+                }
+            }
+        }
+    }
+    
+    // MARK: - Actions
+    
+    @IBAction func addIntake(_ sender: UIBarButtonItem) {
+        let alert = UIAlertController(title: "Add Calorie Intake", message: "Enter the amount of calories", preferredStyle: .alert)
+        var caloriesTextField: UITextField!
+        alert.addTextField { textField in
+            textField.placeholder = "calories"
+            caloriesTextField = textField
+        }
```

Comment:
I know it's easy to forget because the simulator defaults to the hardware keyboard on your Mac, but it is probably best to set the keyboard type for textfields that should only have numeric input.

``` swift
textField.keyboardType = .numberPad
```